File: Lists/c++/Stack.cpp
#include <stdlib.h>
#include <stdio.h>
#include <iostream>
#include "Stack.h"

using namespace std;

class Stack: public AbstractList {
	public:
		SimpleNode *top = NULL;
		SimpleNode *curr = NULL;
		int size = 0;
		Stack(int lenght, int * array) {
			for (int i = 0; i < lenght; i++) {
				push(array[i]);
			}

		}
		void push(int elem) {
			if (!top) {
				top = new SimpleNode();
				top->next = NULL;
				top->data = elem;
			} else {
				curr = new SimpleNode();
				curr->data = elem;
				curr->next = top;
				top = curr;
			}
			size++;
		}
		int pop() {
			if (!top)
				return 0;
			curr = top;
			int value = curr->data;
			top = top->next;
			delete (curr);
			size--;
			return value;

		}

		void printList() {
			curr = top;
			while (curr) {
				cout << curr->data << " ";
				curr = curr->next;
			}
			cout << endl;
		}

		void removeFirstN(int n) {
			for (int i = 0; i < n; i++) {
				pop();
			}
		}

		int getSize() {
			return size;
		}

		int getNth(int n) {
			curr = top;
			int counter = 0;
			while (counter < n) {
				curr = curr->next;
			}
			return curr->data;
		}


		void removeNth(int n){

		
}

		void deleteList(){
			curr = top;
			while(curr){
				curr= top->next;
				delete(top);
				top = curr;
			}	
		}		
};

File: Lists/java/Stack.java
public class Stack<T> {

	public SimpleNode<T> root;
    public int size;

	public T pop(){
		SimpleNode<T> curNode = new SimpleNode<T>();
		T value = root.data;
		curNode = root;
		root = root.next;
		curNode = null;	
        if (size > 0) size--;
		return value;
		
	}
	public T peek(){
		return root.data;
	}
	public int push(T value){
		if (root==null){
			root = new SimpleNode<T>();
			root.data = value;
		}
		else{
			SimpleNode<T> newNode = new SimpleNode<T>();
			newNode.data = value;
			newNode.next = root;
			root = newNode;
		}
        size++;
		return 0;
	}

	public void printAll(){
		if (root!=null){
			SimpleNode<T> curNode  = new SimpleNode<T>();
			curNode = root;
			while(curNode != null){
				System.out.println(pop()+"! ");
				curNode = curNode.next;
			}
		}
	}

	public static void main(String[] args){

		int a[] = {1,2,3,4,5,12};
		Stack<Integer> s = new Stack<Integer>();
		for (int x :a){
			s.push(x);
		}
		s.printAll();
	}
}
